corehq/util/workbook_reading/adapters/csv.py

```diff
@@ -0,0 +1,125 @@
+        else:
+            raise SpreadsheetFileInvalidError(str(error))
+    except IOError as error:
+        raise SpreadsheetFileNotFound(str(error))
+    except FileNotFoundError as error:
+        raise SpreadsheetFileNotFound(str(error))
+
+
+def format_value(str_value):
+    """Attempt to format a str_value into its type."""
+    # Try to see if the str_value is a empty.
+    if str_value == "":
+        return None
+
+    # Try to see if the str_value is a number (float or integer).
+    try:
+        number = ast.literal_eval(str_value)
+        if int(number) == float(number):
+            return int(number)
+        else:
+            return float(number)
+    except (SyntaxError, ValueError):
+        pass
+
+    # Try to see if the str_value is a date or a time.
+    try:
+        parsed_datetime = parse(str_value)
+        # Because dateutil.parser.parse always returns a datetime, we do some guessing
+        # for whether the value is a datetime, a date, or a time.
+        today_midnight = datetime.datetime.now().replace(hour=0, minute=0, second=0, microsecond=0)
+        today_date = today_midnight.date()
+        # If the parsed_datetime time is midnight, then check if the user indicated
+        # midnight or not.
+        if parsed_datetime.time() == today_midnight.time():
+            if ("am" in str_value.lower() or "00:00" in str_value):
+                # The user has indicated to use midnight.
+                # If the str_value is long, then this must be a datetime.
+                if len(str_value) > 10:
+                    return parsed_datetime
+                else:
+                    # The str_value is short, so this must be a time.
+                    return parsed_datetime.time()
+            else:
+                # The user has not indicated to use midnight, so this must be a date.
+                return parsed_datetime.date()
+        elif parsed_datetime.date() == today_date:
+            # We have already handled the case that the user specified a date (but
+            # not a time), so parsed_datetime is a datetime with a time of midnight.
+            # If the parsed_datetime has a date of today, but a time of not midnight,
+            # then either the user specified both the date and the time (meaning
+            # we should return the datetime), or the user specified only the time (so
+            # we should return only the time).
+
+            # If the str_value is long, then this must be a datetime.
+            if len(str_value) > 10:
+                return parsed_datetime
+            else:
+                # The str_value is short, so this must be a time.
+                return parsed_datetime.time()
+        else:
+            # Otherwise, this must be a datetime.
+            return parsed_datetime
+    except ValueError:
+        pass
+
+    # Try to see if the str_value is a boolean.
+    if str_value.lower() in ["true", "t", "yes", "y"]:
```

Comment:
Check out [string_to_boolean](https://github.com/dimagi/commcare-hq/blob/master/corehq/ex-submodules/dimagi/utils/parsing.py#L10) and the other functions in that file.

Reply:
+1 to using an existing convention for the set of values that are considered truthy & falsey.

Reply:
Thanks, that file is helpful!